samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs

```diff
@@ -109,64 +146,327 @@ private MessagingExtensionActionResponse ShareMessageCommand(ITurnContext<IInvok
+                case "webView":
+                    return webView(turnContext, action);
+                case "HTML":
+                    return htmlMethod(turnContext, action);
+                case "razorView":
+                    return razorMethod(turnContext, action);
+                default:
+                    // we are handling two cases within try/catch block 
+                    //if the bot is installed it will create adaptive card attachment and show card with input fields
+                    string memberName;
+                    try
+                    {
+                        // Check if your app is installed by fetching member information.
+                        var member = await TeamsInfo.GetMemberAsync(turnContext, turnContext.Activity.From.Id, cancellationToken);
+                        memberName = member.Name;
+                    }
+                    catch (ErrorResponseException ex)
+                    {
+                        if (ex.Body.Error.Code == "BotNotInConversationRoster")
+                        {
+                            return new MessagingExtensionActionResponse
+                            {
+                                Task = new TaskModuleContinueResponse
+                                {
+                                    Value = new TaskModuleTaskInfo
+                                    {
+                                        Card = GetAdaptiveCardAttachmentFromFile("justintimeinstallation.json"),
+                                        Height = 200,
+                                        Width = 400,
+                                        Title = "Adaptive Card - App Installation",
+                                    },
+                                },
+                            };
+                        }
+                        throw; // It's a different error.
+                    }
                     return new MessagingExtensionActionResponse
                     {
                         Task = new TaskModuleContinueResponse
                         {
                             Value = new TaskModuleTaskInfo
                             {
-                                Card = GetAdaptiveCardAttachmentFromFile("justintimeinstallation.json"),
+                                Card = GetAdaptiveCardAttachmentFromFile("adaptiveCard.json"),
                                 Height = 200,
                                 Width = 400,
-                                Title = "Adaptive Card - App Installation",
+                                Title = $"Welcome {memberName}",
                             },
                         },
                     };
-                }
-                throw; // It's a different error.
             }
+        }
 
-            return new MessagingExtensionActionResponse
+        private MessagingExtensionActionResponse razorMethod(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
```

Comment:
In C#, the convention is use CamelCase for method names.  Please change the first letter of all methods to upper case.